inspirehep/modules/records/serializers/schemas/json.py

```diff
@@ -36,3 +40,105 @@ class RecordSchemaJSONUIV1(Schema):
     links = fields.Raw()
     created = fields.Str()
     updated = fields.Str()
+
+
+class MetadataReferencesSchemaItemV1(Schema):
+    arxiv_eprints = fields.List(fields.Dict())
+    authors = fields.Method('get_authors')
+    collaborations = fields.List(fields.Dict())
+    control_number = fields.Int()
+    dois = fields.List(fields.Dict())
+    publication_info = fields.List(fields.Dict())
+    titles = fields.List(fields.Dict())
+
+    def get_authors(self, data):
+        authors = data.get('authors', [])
+        return authors[:10] or missing
+
+
+class MetadataReferencesSchemaUIV1(Schema):
+    references = fields.Nested(
+        MetadataReferencesSchemaItemV1, many=True, dump_only=True)
+
+
+class ReferencesSchemaJSONUIV1(RecordSchemaJSONUIV1):
+    """Schema for references."""
+    metadata = fields.Nested(MetadataReferencesSchemaUIV1, dump_only=True)
+
+    @pre_dump
+    def pre_process(self, data):
+        resolved = []
+        references = data['metadata'].get('references')
+        if references:
+            reference_records = self.resolve_records(references)
+
+            for reference in references:
+                if 'record' in reference:
+                    reference_record_id = get_recid_from_ref(
+                        reference.get('record'))
+                    reference_record = reference_records.get(
+                        reference_record_id)
+                    if reference_record:
+                        resolved.append(reference_record)
+                        break
+
+                reference = self.format_references(reference)
+                if reference:
+                    resolved.append(reference)
+
+        data['metadata']['references'] = resolved
+        return data
+
+    def resolve_records(self, references):
+        ids = [
+            get_recid_from_ref(reference['record'])
+            for reference in references if 'record' in reference
+        ]
+        resolved_records = get_resolved_references(ids)
+        return {
+            record['control_number']: record
+            for record in resolved_records
+        }
+
+    def format_references(self, reference):
+        reference.update(reference.get('reference', {}))
+        reference.pop('reference', None)
+        reference.update({
+            'publication_info': self.prepare_publication_info(
+                reference.get('publication_info', {})),
+            'arxiv_eprints': self.prepare_arxiv_eprint(
+                reference.get('arxiv_eprint')),
+            'collaborations': self.prepare_collaborations(
+                reference.get('collaborations', [])),
+            'dois': self.prepare_dois(reference.get('dois', [])),
+            'titles': self.prepare_titles(reference.get('title', [])),
+        })
+        reference = strip_empty_values(reference)
+        return self.only_meaningful_data(reference)
+
+    def only_meaningful_data(self, reference):
```

Comment:
* as discussed IRL, I think it's strange if you don't silently skip references. We probably need to design more generally how to show unlinked references, but that can be done later.
* this and all next methods can be decorated with `@staticmethod`.